src/main/java/bankservice/demo/service/TransactionServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package bankservice.demo.service;
+
+import bankservice.demo.entity.Account;
+import bankservice.demo.entity.Transaction;
+import bankservice.demo.repository.AccountRepository;
+import bankservice.demo.repository.TransactionRepository;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TransactionServiceImpl implements TransactionService {
+    private final TransactionRepository transactionRepository;
+    private final AccountRepository accountService;
+
+    public TransactionServiceImpl(TransactionRepository transactionRepository,
+                                  AccountRepository accountService) {
+        this.transactionRepository = transactionRepository;
+        this.accountService = accountService;
+    }
+
+    @Override
+    public List<Transaction> getAllByAccount(int page, int size, Account account) {
+        PageRequest pageRequest = PageRequest.of(page, size,
+                Sort.by("date").descending().and(Sort.by("id")));
+        return transactionRepository.getAllByAccount(account, pageRequest);
+    }
+
+    @Override
+    public void transfer(Account fromAccount, Account toAccount, BigDecimal amount) {
```

Comment:
it would be great to use `org.springframework.transaction.annotation.Transactional` here above the method

see more https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Transactional.html